confmap/expand.go

```diff
@@ -76,30 +76,42 @@ func (mr *Resolver) expandValue(ctx context.Context, value any) (any, bool, erro
 	return value, false, nil
 }
 
+func (mr *Resolver) findURIHelper(remaining string) (string, string) {
+	// if remaining does not contain "}", there are no URIs left: stop recursion.
+	if !strings.Contains(remaining, "}") {
+		return "", ""
+	}
+	return mr.findURI(remaining)
+}
+
 // findURI attempts to find the first expandable URI in input. It returns an expandable
 // URI, or an empty string if none are found.
 // Note: findURI is only called when input contains a closing bracket.
-func findURI(input string) string {
+func (mr *Resolver) findURI(input string) (string, string) {
```

Comment:
Instead of changing the `findURI` logic that much, I would rather change it to return all "expandable" even if no schema, and in the expandUri would add the logic if no schema and a defaultSchema present do that, otherwise do nothing and continue to expand.

Reply:
I agree with reverting back to my previous implementation that does not do any magic scheme appending in `findURI` and doing more default logic in other functions.  I'll work on this.

Reply:
@bogdandrutu I believe I've addressed all your feedback, ready for another review.